pkg/ipamd/ipamd.go

```diff
@@ -1691,6 +1715,80 @@ func (c *IPAMContext) warmIPTargetsDefined() bool {
 	return c.warmIPTarget != noWarmIPTarget || c.minimumIPTarget != noMinimumIPTarget
 }
 
+// max pods from instance type mapping file
+type instanceTypeMaxPodsMapping map[string]int64
+
+// getMaxPodsFromFile reads the max pods value from the eni-max-pods.txt file
+// based on the instance type
+func (c *IPAMContext) getMaxPodsFromFile() (int64, error) {
+	instanceType := c.awsClient.GetInstanceType()
+	if instanceType == "" {
+		return 0, fmt.Errorf("failed to get instance type")
+	}
+
+	data, err := os.ReadFile(eniMaxPodsFilePath)
+	if err != nil {
+		return 0, fmt.Errorf("failed to read ENI max pods file: %w", err)
+	}
+
+	mapping := parseMaxPodsFile(string(data))
```

Comment:
Looks like this call is made on node base, which mean this ipamd only need to know how many pods this node can support. Returning a map seems unnecessary?

Reply:
makes sense, will modify this func to return maxPods val directly